services/nmma_analysis_service/app.py

```diff
@@ -0,0 +1,354 @@
+    # locally and then write their contents
+    # to the results dictionary for uploading
+    local_temp_files = []
+    cand_name = data_dict['resource_id']
+
+    prior_directory = f"{os.path.dirname(os.path.realpath(__file__))}/priors"
+    svdmodel_directory = f"{os.path.dirname(os.path.realpath(__file__))}/svdmodels"
+
+    try:
+        ##########################
+        # Setup parameters and fit
+        ##########################
+        plotdir = tempfile.mkdtemp()
+
+        # cpus = 2
+        nlive = 32
+        error_budget = 1.0
+
+        tmin = 0.01
+        tmax = 7
+        dt = 0.1
+        Ebv_max = 0.5724
+
+        t0 = np.min(data['mjd'])
+        interpolation_type = "sklearn_gp"
+        sampler = "pymultinest"
+
+        prior = f"{prior_directory}/{source}.prior"
+        if not os.path.isfile(prior):
+            log(f"Prior file for model {source} does not exist")
+            return
+        priors = bilby.gw.prior.PriorDict(prior)
+        if fix_z:
+            if z is not None:
+                from astropy.cosmology import Planck18 as cosmo
+
+                priors['luminosity_distance'] = cosmo.luminosity_distance(z).value
+            else:
+                raise ValueError("No redshift provided but `fix_z` requested.")
+        priors.to_file(plotdir, source)
+        prior = os.path.join(plotdir, f'{source}.prior')
+        local_temp_files.append(prior)
+
+        # output the data
+        # in the format desired by NMMA
+        f = tempfile.NamedTemporaryFile(suffix=".dat", mode="w", delete=False)
+        for row in data:
+            tt = Time(row['mjd'], format='mjd').isot
+            filt = row['filter'][-1]
+            mag = row['mag']
+            magerr = row['magerr']
+            f.write(f"{tt} {filt} {mag} {magerr}\n")
+        f.close()
+        local_temp_files.append(f.name)
+
+        # NMMA lightcurve fitting
+        # triggered with a shell command
+        command = subprocess.run(
+            "light_curve_analysis"
+            + " --model "
+            + source
+            + " --svd-path "
+            + svdmodel_directory
+            + " --outdir "
+            + plotdir
+            + " --label "
+            + cand_name
+            + "_"
+            + source
+            + " --trigger-time "
+            + str(t0)
+            + " --data "
+            + f.name
+            + " --prior "
+            + prior
+            + " --tmin "
+            + str(tmin)
+            + " --tmax "
+            + str(tmax)
+            + " --dt "
+            + str(dt)
+            + " --error-budget "
+            + str(error_budget)
+            + " --nlive "
+            + str(nlive)
+            + " --Ebv-max "
+            + str(Ebv_max)
+            + " --interpolation_type "
+            + interpolation_type
+            + " --sampler "
+            + sampler
+            + " --plot",
+            shell=True,
+            capture_output=True,
+        )
```

Comment:
If we have to call an executable we need to make sure there are no harmful expressions. 
For example if a user tries to look for a source called `ZTFabcd113 && rm -rf /` we could get into some trouble. 
One way would be to make sure there are no illegal characters on any user-input. I'm not sure if this covers us 100% though. I think probably @profjsb and @stefanv would want to weigh in since this could be a critical security issue.

Reply:
Yeah this is a good point. I am removing this for now, as this should be dealt with in another PR.